zone/gm_commands/parcels.cpp

```diff
@@ -4,304 +4,488 @@
-			parcel_out.from_name = c->GetName();
-			parcel_out.note      = note;
-			parcel_out.sent_date = time(nullptr);
-			parcel_out.quantity  = quantity == 0 ? 1 : quantity;
-			parcel_out.item_id   = PARCEL_MONEY_ITEM_ID;
-			parcel_out.char_id   = send_to_client.at(0).char_id;
-			parcel_out.slot_id   = next_slot;
-			parcel_out.id        = 0;
-
-			auto result = CharacterParcelsRepository::InsertOne(database, parcel_out);
+			CharacterParcelsRepository::CharacterParcels p;
+			p.from_name = c->GetName();
+			p.note      = note;
+			p.sent_date = time(nullptr);
+			p.quantity  = quantity == 0 ? 1 : quantity;
+			p.item_id   = PARCEL_MONEY_ITEM_ID;
+			p.char_id   = character_id;
+			p.slot_id   = free_slot;
+			p.id        = 0;
+
+			auto result = CharacterParcelsRepository::InsertOne(database, p);
 			if (!result.id) {
 				LogError(
-					"Failed to add parcel to database.  From {} to {} item {} quantity {}",
-					parcel_out.from_name,
-					send_to_client.at(0).character_name,
-					parcel_out.item_id,
-					parcel_out.quantity
+					"Failed to add parcel to database. From {} to {} item {} with a quantity of {}.",
+					p.from_name,
+					character_name,
+					p.item_id,
+					p.quantity
 				);
 				c->Message(
-					Chat::Yellow,
+					Chat::White,
 					"Unable to save parcel to the database. Please contact an administrator."
 				);
 				return;
 			}
 
-			c->MessageString(
-				Chat::Yellow,
-				PARCEL_DELIVERY,
-				c->GetCleanName(),
-				"Money",
-				send_to_client.at(0).character_name.c_str()
+			c->Message(
+				Chat::White,
+				fmt::format(
+					"Successfully sent {} Copper to {}{}.",
+					Strings::Commify(quantity),
+					character_name,
+					note_string
+				).c_str()
 			);
 
 			if (player_event_logs.IsEventEnabled(PlayerEvent::PARCEL_SEND)) {
-				PlayerEvent::ParcelSend e{};
-				e.from_player_name = parcel_out.from_name;
-				e.to_player_name   = send_to_client.at(0).character_name;
-				e.item_id          = parcel_out.item_id;
-				e.quantity         = parcel_out.quantity;
-				e.sent_date        = parcel_out.sent_date;
+				PlayerEvent::ParcelSend e{ };
+				e.from_player_name = p.from_name;
+				e.to_player_name   = character_name;
+				e.item_id          = p.item_id;
+				e.quantity         = p.quantity;
+				e.sent_date        = p.sent_date;
 
 				RecordPlayerEventLogWithClient (c, PlayerEvent::PARCEL_SEND, e);
 			}
 
-			Parcel_Struct ps{};
-			ps.item_slot = parcel_out.slot_id;
-			strn0cpy(ps.send_to, send_to_client.at(0).character_name.c_str(), sizeof(ps.send_to));
+			Parcel_Struct ps{ };
+			ps.item_slot = p.slot_id;
+			strn0cpy(ps.send_to, character_name.c_str(), sizeof(ps.send_to));
 
 			c->SendParcelDeliveryToWorld(ps);
-		}
-		else {
-			auto item = database.GetItem(item_id);
+		} else {
+			auto item = database.CreateItem(
+				item_id,
+				quantity,
+				augment_ids[0],
+				augment_ids[1],
+				augment_ids[2],
+				augment_ids[3],
+				augment_ids[4],
+				augment_ids[5]
+			);
```

Comment:
With the change to create item, item will need to be delete at all the right places or it will cause a leak I believe.  I would suggest using a unique_ptr so that you don't need to worry about the deletes.